test/terra/test_noise_model.py

```diff
@@ -517,6 +518,73 @@ def test_amplitude_damping_error(self):
         self.is_completed(result)
         self.compare_counts(result, [circuit], [target], delta=0.05 * shots)
 
+    def test_standard_reset0_error_100percent(self):
+        """Test 100% Pauli error on id gates"""
+        qr = QuantumRegister(1, 'qr')
+        cr = ClassicalRegister(1, 'cr')
+        circuit = QuantumCircuit(qr, cr)
+        circuit.x(qr)
+        circuit.barrier(qr)
+        circuit.measure(qr, cr)
+        backend = QasmSimulator()
+        shots = 100
+        # test noise model
+        error = reset_error(1)
+        noise_model = NoiseModel()
+        noise_model.add_all_qubit_quantum_error(error, ['id', 'x'])
+        # Execute
+        target = {'0x0': shots}
+        qobj = compile([circuit], backend, shots=shots,
+                       basis_gates=noise_model.basis_gates)
+        result = backend.run(qobj, noise_model=noise_model).result()
+        self.is_completed(result)
+        self.compare_counts(result, [circuit], [target], delta=0)
```

Comment:
Maybe as a TODO, but a test should always expect any kind of `assert` idiom, so instead of having a `self.compare_counts()` I'd rather have, for example, a `self.assertCountsEqual()`